pkg/api/pod/util.go

```diff
@@ -813,3 +818,50 @@ func setsWindowsHostProcess(podSpec *api.PodSpec) bool {
 
 	return inUse
 }
+
+func ValidatePodTemplateSpecAllowInValidLabelValueInSelector(spec *api.PodSpec) bool {
+	if spec.Affinity != nil {
+		return ValidateAffinityAllowInValidLabelValueInSelector(spec.Affinity)
+	}
+	return false
+}
+
+func ValidateAffinityAllowInValidLabelValueInSelector(affinity *core.Affinity) bool {
+	if affinity.PodAffinity != nil {
+		if affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+			for _, term := range affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
+				allErrs := apivalidation.ValidatePodAffinityTermSelector(term, false, field.NewPath("RequiredDuringSchedulingIgnoredDuringExecution"))
```

Comment:
pass nil for fieldpath, since we're not using the path errors (applies to all the places we're just checking validity without using the returned error content)

Reply:
I think the method `ValidateLabelSelector` we use has the `fldPath.Child` logic to keep the error path. 

But in `Path.Child`, there is any nil-check here, so if we pass the nil to the method, I think there will be some NPE panic